src/core/player.cpp

```diff
@@ -213,7 +215,25 @@ void Player::NextItem(Engine::TrackChangeFlags change) {
   // Manual track changes override "Repeat track"
   const bool ignore_repeat_track = change & Engine::Manual;
 
-  int i = active_playlist->next_row(ignore_repeat_track);
+  int i = active_playlist->current_row();
+  int original_index = i;
```

Comment:
You could move this into the scope of the second `if`.

Reply:
To be fixed in the next commit.